### ISSUE TITLE
Misaligned returns when parameters are split

### ISSUE THREAD

**Issue by fvictorio** [prettier-solidity/prettier-plugin-solidity#116]
This

```solidity
contract Foo {
  function foo() modifier1 modifier2 aLongModifier anotherVeryLongModifier returns (uint aVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongParameterName) {
    a = 42;
  }
}
```

is prettified as:

```solidity
contract Foo {
  function foo()
    modifier1
    modifier2
    aLongModifier
    anotherVeryLongModifier
    returns (
    uint aVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongParameterName
  )
  {
    a = 42;
  }
}
```

Two things are wrong here: the indentation of the returns parameter and closing parentheses, and that the closing parentheses and the opening body brace are not in the same line.

**Comment by Janther** [comment]
I've been working on this for a while and I'd like to talk about the position of the opening body brace.

This is how it is now:
```Solidity
contract Foo {
    function foo()
        modifier1
        modifier2
        aLongModifier
        anotherVeryLongModifier
        returns (
        uint aVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongParameterName
    )
    {
        a = 42;
    }
}
```
Fixing the indentation it looks like this:

```Solidity
contract Foo {
    function foo()
        modifier1
        modifier2
        aLongModifier
        anotherVeryLongModifier
        returns (
            uint aVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongParameterName
        )
    {
        a = 42;
    }
}
```
Now. if we want the opening body in the same line as the closing parenthesys, it will look like this
```Solidity
contract Foo {
    function foo()
        modifier1
        modifier2
        aLongModifier
        anotherVeryLongModifier
        returns (
            uint aVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongParameterName
        ) {
        a = 42;
    }
}
```
now there is no clear separation between the body and the function signature.

there is another option to have the return statement unindented like this:

```Solidity
contract Foo {
    function foo()
        modifier1
        modifier2
        aLongModifier
        anotherVeryLongModifier
    returns (
        uint aVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongParameterName
    ) {
        a = 42;
    }
}
```

I'd like us to discuss which is the way to go with this one.

**Comment by fvictorio** [comment]
The second option is the one that looks better, at least with this example, but it's not consistent with how it's being done in other scenarios. Option 3 would be the consistent choice IMO, but I agree it doesn't look nice. Finally, option 4 looks nice (after a while; at first it looks awful); I like how it separates modifiers from the returned values. But I'm afraid it might be too "different".

If I had to choose right now, I would go with option 3 and reconsider it later with more user feedback.

**Comment by Janther** [comment]
I just made a PR for this issue.
however after looking at it there are side effects I don't like about solution 3.
for example this test:
```Solidity
contract AddressPayable {
    address payable[] hello;
    function sendSomeEth(address payable to, address payable[] memory world)
        public
        payable {
        address payable target = to;
        target.transfer(msg.value);
    }
}
```
that should look like this:
```Solidity
contract AddressPayable {
    address payable[] hello;
    function sendSomeEth(address payable to, address payable[] memory world)
        public
        payable
    {
        address payable target = to;
        target.transfer(msg.value);
    }
}
```

so the solution that I'm more comfortable with and doesn't have this side effect would be number 2

```Solidity
contract Foo {
    function foo()
        modifier1
        modifier2
        aLongModifier
        anotherVeryLongModifier
        returns (
            uint aVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongParameterName
        )
    {
        a = 42;
    }
}
```

I can add an extra commit to #129 that will fix this.
please have a look at this.
@fvictorio @mattiaerre

**Comment by mattiaerre** [comment]
this is awesome, thanks @Janther please feel free to add another commit to #129 
thanks,
-Mattia

**Comment by Janther** [comment]
done, please have a look

**Comment by mattiaerre** [comment]
closed by #129
